sdk/servicebus/azure-servicebus/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,1335 @@
+
+**SessionCannotBeLockedError resolution:**
+1. Ensure no other clients are already connected to the same session
+2. Wait for the current session lock to expire before reconnecting
+3. Use a different session ID if specific session is not required
+
+## Troubleshooting sender issues
+
+### Cannot send batch with multiple partition keys
+
+When sending to a partition-enabled entity, all messages included in a single send operation must have the same `session_id` if the entity is session-enabled, or the same custom properties that determine partitioning.
+
+**Error symptoms:**
+- Messages are rejected or go to different partitions than expected
+- Inconsistent message ordering
+
+**Resolution:**
+1. **For session-enabled entities, ensure all messages in a batch have the same session ID:**
+```python
+from azure.servicebus import ServiceBusMessage
+
+# Correct: All messages have the same session_id
+messages = [
+    ServiceBusMessage("Message 1", session_id="session1"),
+    ServiceBusMessage("Message 2", session_id="session1"),
+    ServiceBusMessage("Message 3", session_id="session1")
+]
+
+with sender:
+    sender.send_messages(messages)
+```
+
+2. **For partitioned entities, group messages by partition key:**
+```python
+# Group messages by partition key before sending
+partition1_messages = [
+    ServiceBusMessage("Message 1", application_properties={"region": "east"}),
+    ServiceBusMessage("Message 2", application_properties={"region": "east"})
+]
+
+partition2_messages = [
+    ServiceBusMessage("Message 3", application_properties={"region": "west"}),
+    ServiceBusMessage("Message 4", application_properties={"region": "west"})
+]
+
+# Send each group separately
+with sender:
+    sender.send_messages(partition1_messages)
+    sender.send_messages(partition2_messages)
+```
+
+### Batch fails to send
+
+The Service Bus service has size limits for message batches and individual messages.
+
+**Error symptoms:**
+- `MessageSizeExceededError` when sending batches
+- Messages larger than expected failing to send
+
+**Resolution:**
+1. **Reduce batch size or message payload:**
+```python
+from azure.servicebus import ServiceBusMessage
+from azure.servicebus.exceptions import MessageSizeExceededError
+import json
+
+def send_large_dataset(sender, data_list, max_batch_size=100):
+    """Send large datasets in smaller batches"""
+    for i in range(0, len(data_list), max_batch_size):
+        batch = data_list[i:i + max_batch_size]
+        messages = [ServiceBusMessage(json.dumps(item)) for item in batch]
+        
+        try:
+            sender.send_messages(messages)
+        except MessageSizeExceededError:
+            # If batch is still too large, send individually
+            for message in messages:
+                sender.send_messages(message)
+```
+
+2. **Check message size limits:**
+   - Standard tier: 256 KB per message
+   - Premium tier: 1 MB per message
+   - Batch limit: 1 MB regardless of tier
```

Comment:
Remove this section and ask users to go here to get the latest batch size limits https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-quotas